Error while checking build/O2/o2-dataflow for cb21da7133f4d6ed802d7a29b8194effc8ddbb79 at 2022-09-12 14:51:
```
## sw/BUILD/O2-latest/log
100% tests passed, 0 tests failed out of 426
 94/106 Test #108: test_Framework_test_SingleDataSource ....................***Failed    2.65 sec
[ERROR] pid 25126 (internal-dpl-injected-dummy-sink) crashed with 1
[ERROR] SEVERE: Device internal-dpl-injected-dummy-sink (25126) returned with 1
99% tests passed, 1 tests failed out of 101
```
Full log [here](https://ali-ci.cern.ch/alice-build-logs/AliceO2Group/AliceO2/9796/cb21da7133f4d6ed802d7a29b8194effc8ddbb79/build_O2_o2-dataflow/pretty.html).

--- a/Detectors/EMCAL/workflow/include/EMCALWorkflow/PublisherSpec.h
+++ b/Detectors/EMCAL/workflow/include/EMCALWorkflow/PublisherSpec.h
@@ -46,7 +46,7 @@ struct PublisherConf {
 };
 
 template <typename T = void>
-framework::DataProcessorSpec getPublisherSpec(PublisherConf const& config, bool propagateMC = true)
+framework::DataProcessorSpec getPublisherSpec(PublisherConf const& config, uint32_t subspec = 0, bool propagateMC = true)
 {
   using Reader = o2::framework::RootTreeReader;
   using Output = o2::framework::Output;
@@ -56,28 +56,28 @@ framework::DataProcessorSpec getPublisherSpec(PublisherConf const& config, bool 
   auto mco = o2::framework::DataSpecUtils::asConcreteDataTypeMatcher(config.mcoutput);
 
   // a creator callback for the actual reader instance
-  auto creator = [dto, tro, mco, propagateMC](const char* treename, const char* filename, int nofEvents, Reader::PublishingMode publishingMode, const char* branchname, const char* triggerbranchname, const char* mcbranchname) {
+  auto creator = [dto, tro, mco, subspec, propagateMC](const char* treename, const char* filename, int nofEvents, Reader::PublishingMode publishingMode, const char* branchname, const char* triggerbranchname, const char* mcbranchname) {
     constexpr auto persistency = o2::framework::Lifetime::Timeframe;
     if (propagateMC) {
       return std::make_shared<Reader>(treename,
                                       filename,
                                       nofEvents,
                                       publishingMode,
-                                      Output{mco.origin, mco.description, 0, persistency},
+                                      Output{mco.origin, mco.description, subspec, persistency},
                                       mcbranchname,
-                                      Reader::BranchDefinition<T>{Output{dto.origin, dto.description, 0, persistency}, branchname},
-                                      Reader::BranchDefinition<TriggerInputType>{Output{tro.origin, tro.description, 0, persistency}, triggerbranchname});
+                                      Reader::BranchDefinition<T>{Output{dto.origin, dto.description, subspec, persistency}, branchname},
+                                      Reader::BranchDefinition<TriggerInputType>{Output{tro.origin, tro.description, subspec, persistency}, triggerbranchname});
     } else {
       return std::make_shared<Reader>(treename,
                                       filename,
                                       nofEvents,
                                       publishingMode,
-                                      Reader::BranchDefinition<T>{Output{dto.origin, dto.description, 0, persistency}, branchname},
-                                      Reader::BranchDefinition<TriggerInputType>{Output{tro.origin, tro.description, 0, persistency}, triggerbranchname});
+                                      Reader::BranchDefinition<T>{Output{dto.origin, dto.description, subspec, persistency}, branchname},
+                                      Reader::BranchDefinition<TriggerInputType>{Output{tro.origin, tro.description, subspec, persistency}, triggerbranchname});
     }
   };
 
-  return createPublisherSpec(config, propagateMC, creator);
+  return createPublisherSpec(config, subspec, propagateMC, creator);
 }
 
 inline framework::DataProcessorSpec getCellReaderSpec(bool propagateMC)
@@ -92,6 +92,7 @@ inline framework::DataProcessorSpec getCellReaderSpec(bool propagateMC)
                                                        o2::framework::OutputSpec{"EMC", "CELLS"},
                                                        o2::framework::OutputSpec{"EMC", "CELLSTRGR"},
                                                        o2::framework::OutputSpec{"EMC", "CELLSMCTR"}},
+                                         0,
                                          propagateMC);
 }
 
@@ -101,7 +102,7 @@ using Reader = o2::framework::RootTreeReader;
 using Creator = std::function<std::shared_ptr<Reader>(const char*, const char*, int, Reader::PublishingMode, const char*, const char*, const char*)>;
 } // namespace workflow_reader
 
-framework::DataProcessorSpec createPublisherSpec(PublisherConf const& config, bool propagateMC, workflow_reader::Creator creator);
+framework::DataProcessorSpec createPublisherSpec(PublisherConf const& config, uint32_t subspec, bool propagateMC, workflow_reader::Creator creator);
 
 } // namespace emcal
 } // end namespace o2

--- a/Detectors/EMCAL/workflow/src/PublisherSpec.cxx
+++ b/Detectors/EMCAL/workflow/src/PublisherSpec.cxx
@@ -23,7 +23,7 @@ namespace o2
 namespace emcal
 {
 
-o2::framework::DataProcessorSpec createPublisherSpec(PublisherConf const& config, bool propagateMC, workflow_reader::Creator creator)
+o2::framework::DataProcessorSpec createPublisherSpec(PublisherConf const& config, uint32_t subspec, bool propagateMC, workflow_reader::Creator creator)
 {
   struct ProcessAttributes {
     std::shared_ptr<o2::framework::RootTreeReader> reader;
@@ -32,7 +32,7 @@ o2::framework::DataProcessorSpec createPublisherSpec(PublisherConf const& config
     bool finished;
   };
 
-  auto initFunction = [config, propagateMC, creator](o2::framework::InitContext& ic) {
+  auto initFunction = [config, subspec, propagateMC, creator](o2::framework::InitContext& ic) {
     // get the option from the init context
     auto filename = o2::utils::Str::concat_string(o2::utils::Str::rectifyDirectory(ic.options().get<std::string>("input-dir")),
                                                   ic.options().get<std::string>("infile"));
@@ -84,15 +84,15 @@ o2::framework::DataProcessorSpec createPublisherSpec(PublisherConf const& config
     return processFunction;
   };
 
-  auto createOutputSpecs = [&config, propagateMC]() {
+  auto createOutputSpecs = [&config, subspec, propagateMC]() {
     std::vector<o2::framework::OutputSpec> outputSpecs;
     auto dto = o2::framework::DataSpecUtils::asConcreteDataTypeMatcher(config.dataoutput);
     auto tro = o2::framework::DataSpecUtils::asConcreteDataTypeMatcher(config.triggerrecordoutput);
     auto mco = o2::framework::DataSpecUtils::asConcreteDataTypeMatcher(config.mcoutput);
-    outputSpecs.emplace_back(o2::framework::OutputSpec{{"output"}, dto.origin, dto.description, 0, o2::framework::Lifetime::Timeframe});
-    outputSpecs.emplace_back(o2::framework::OutputSpec{{"outputTRG"}, tro.origin, tro.description, 0, o2::framework::Lifetime::Timeframe});
+    outputSpecs.emplace_back(o2::framework::OutputSpec{{"output"}, dto.origin, dto.description, subspec, o2::framework::Lifetime::Timeframe});
+    outputSpecs.emplace_back(o2::framework::OutputSpec{{"outputTRG"}, tro.origin, tro.description, subspec, o2::framework::Lifetime::Timeframe});
     if (propagateMC) {
-      outputSpecs.emplace_back(o2::framework::OutputSpec{{"outputMC"}, mco.origin, mco.description, 0, o2::framework::Lifetime::Timeframe});
+      outputSpecs.emplace_back(o2::framework::OutputSpec{{"outputMC"}, mco.origin, mco.description, subspec, o2::framework::Lifetime::Timeframe});
     }
     return std::move(outputSpecs);
   };

--- a/Detectors/EMCAL/workflow/src/RecoWorkflow.cxx
+++ b/Detectors/EMCAL/workflow/src/RecoWorkflow.cxx
@@ -132,7 +132,7 @@ o2::framework::WorkflowSpec getWorkflow(bool propagateMC,
                                                                        o2::framework::OutputSpec{"EMC", "DIGITS"},
                                                                        o2::framework::OutputSpec{"EMC", "DIGITSTRGR"},
                                                                        o2::framework::OutputSpec{"EMC", "DIGITSMCTR"}},
-                                                                     propagateMC));
+                                                                     0, propagateMC));
     }
 
     if (enableDigitsPrinter) {
@@ -155,7 +155,7 @@ o2::framework::WorkflowSpec getWorkflow(bool propagateMC,
                                                                       o2::framework::OutputSpec{"EMC", "CELLS"},
                                                                       o2::framework::OutputSpec{"EMC", "CELLSTRGR"},
                                                                       o2::framework::OutputSpec{"EMC", "CELLSMCTR"}},
-                                                                    propagateMC));
+                                                                    0, propagateMC));
     }
     if (enableDigitsPrinter) {
       try {

--- a/Detectors/EMCAL/workflow/src/cell-reader-workflow.cxx
+++ b/Detectors/EMCAL/workflow/src/cell-reader-workflow.cxx
@@ -44,6 +44,7 @@ WorkflowSpec defineDataProcessing(ConfigContext const& cfgc)
   bool disableMC = cfgc.options().get<bool>("disable-mc");
   auto subspec = cfgc.options().get<uint32_t>("subspec");
   o2::conf::ConfigurableParam::updateFromString(cfgc.options().get<std::string>("configKeyValues"));
+  LOG(info) << "Cell reader: publishing on subspec " << subspec << std::endl;
 
   WorkflowSpec specs;
   specs.emplace_back(o2::emcal::getPublisherSpec<std::vector<o2::emcal::Cell>>(PublisherConf{
@@ -53,10 +54,10 @@ WorkflowSpec defineDataProcessing(ConfigContext const& cfgc)
                                                                                  {"cellbranch", "EMCALCell", "Cell branch"},
                                                                                  {"celltriggerbranch", "EMCALCellTRGR", "Trigger record branch"},
                                                                                  {"mcbranch", "EMCALCellMCTruth", "MC label branch"},
-                                                                                 o2::framework::OutputSpec{"EMC", "CELLS", subspec},
-                                                                                 o2::framework::OutputSpec{"EMC", "CELLSTRGR", subspec},
-                                                                                 o2::framework::OutputSpec{"EMC", "CELLSMCTR", subspec}},
-                                                                               !disableMC));
+                                                                                 o2::framework::OutputSpec{{"cells"}, "EMC", "CELLS", subspec, o2::framework::Lifetime::Timeframe},
+                                                                                 o2::framework::OutputSpec{{"triggerrecords"}, "EMC", "CELLSTRGR", subspec, o2::framework::Lifetime::Timeframe},
+                                                                                 o2::framework::OutputSpec{{"mclabels"}, "EMC", "CELLSMCTR", subspec, o2::framework::Lifetime::Timeframe}},
+                                                                               subspec, !disableMC));
 
   // configure dpl timer to inject correct firstTForbit: start from the 1st orbit of TF containing 1st sampled orbit
   o2::raw::HBFUtilsInitializer hbfIni(cfgc, specs);

